src/ray/gcs/redis_context.cc

```diff
@@ -598,55 +670,53 @@ Status RedisContext::Connect(const std::string &address,
 
   RAY_CHECK(!context_);
   RAY_CHECK(!redis_async_context_);
+
+  bool should_reconnect = true;
+
+  // Remember function arguments for reconnection
+  if (is_first_connect_) {
+    is_first_connect_ = false;
+    address_ = address;
+    port_ = port;
+    username_ = username;
+    password_ = password;
+    enable_ssl_ = enable_ssl;
+    // Don't try to reconnect for the first time.
```

Comment:
why?

Reply:
This was used to prevent infinite reconnection during startup because the failure in the first connection is usually due to user misconfiguration of the Redis server, such as an incorrect password. However, in https://github.com/ray-project/ray/pull/48781#discussion_r1887153766, I've changed it to a fatal error in the `Connect` function if we fail to connect to the saved `RAY_REDIS_ADDRESS`. Therefore, this is no longer needed.